lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -90,6 +90,14 @@
             - To detach a disk from a vm, explicitly set to ''.
             - If this option is unset, the value will not be changed.
         version_added: 2.5
+    caching:
+        description:
+            - "Disk caching policy."
+            - "Allowed values: None, ReadOnly, ReadWrite."
+        choices:
+            - None
```

Comment:
we should have pythonized values here: [read_only, read_write]
Also it's better not to use name "None" or "none".
if there's  a need to explicitly disable caching, we should think of some other name, for instance "disabled".

Reply:
Compared with `disabled`, I would like to use an empty string str('') to disable this option, like the `managed_by` option

Reply:
the only difference by **managed_by** is that' it's actually resource id, and this option is a choice, so I think we would have to explicitly define empty choice {'', 'read_only', 'read_write'}, otherwise Ansible won't accept empty string as a value.

Reply:
There are three states "None", "Read-only" and "Read/write", which are shown on the azure portal for disk caching . The default state is "None". In the azure-python-sdk, there is an enum type called CachingTypes. It includes three fields none, read_only and read_write.

some selections are listed for discussion,
- In consistent with az cli command, we can define {'none', 'read_only', 'read_write'}
- Because the default state is "None", we can define {'read_only','read_write'}
- explicitly define the empty choice like zim's suggestion

Do you think which one is the best choice?

Reply:
I vote 3

For `None`, this means default when creation, nothing happen when update.
For `str('')`, indicates set this option to `None` whenever.

Reply:
@zikalino @yuwzho I have pushed a new commit. Please help me review it again. Thank you all very much.